lib/galaxy/tools/parameters/basic.py

```diff
@@ -1361,9 +1361,12 @@ def get_legal_values(self, trans, other_values):
         legal_values = self.get_column_list(trans, other_values)
 
         value = other_values.get(self.name)
-        if value is not None and value not in legal_values and self.is_file_empty(trans, other_values):
-            value = value if isinstance(value, list) else [value]
-            legal_values.extend(value)
+        if value is not None:
+            # There are cases where 'value' is a string of comma separated values
+            # This ensures that it is converted into a list
+            value = value if isinstance(value, list) else value.split(",")
```

Comment:
Or you could call `galaxy.util.listify()` https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/util/__init__.py#L967 - it does just that, but also handles tuples, so is a bit more generic.

Reply:
Good idea. Changed accordingly.

Reply:
Looks good to me, thank you!